test/skimNtuple2018_HHbtag.cpp

```diff
@@ -2483,6 +2536,96 @@ int main (int argc, char** argv)
     theSmallTree.m_dau1_byTightCombinedIsolationDeltaBetaCorr3Hits = ( theBigTree.tauID->at (firstDaughterIndex)  & (1 << ibit) ) ? true : false ;
     theSmallTree.m_dau2_byTightCombinedIsolationDeltaBetaCorr3Hits = ( theBigTree.tauID->at (secondDaughterIndex) & (1 << ibit) ) ? true : false ;
 
+    if(isHHsignal){
+      theSmallTree.m_genH1_pt = vH1.Pt();
+      theSmallTree.m_genH1_eta = vH1.Eta();
+      theSmallTree.m_genH1_phi = vH1.Phi();
+      theSmallTree.m_genH1_e = vH1.E();
+      theSmallTree.m_genH2_pt = vH2.Pt();
+      theSmallTree.m_genH2_eta = vH2.Eta();
+      theSmallTree.m_genH2_phi = vH2.Phi();
+      theSmallTree.m_genH2_e = vH2.E();
+
+      theSmallTree.m_genB1_pt = vGenB1.Pt();
+      theSmallTree.m_genB1_eta = vGenB1.Eta();
+      theSmallTree.m_genB1_phi = vGenB1.Phi();
+      theSmallTree.m_genB1_e = vGenB1.E();
+      theSmallTree.m_genB2_pt = vGenB2.Pt();
+      theSmallTree.m_genB2_eta = vGenB2.Eta();
+      theSmallTree.m_genB2_phi = vGenB2.Phi();
+      theSmallTree.m_genB2_e = vGenB2.E();
```

Comment:
Hi @kramerto ! I have a question here: why do we want to store the generated b qaurk information only for the signals? If we want to perform efficiency studies of b-tagging algorithms we need to know which are the generated b quarks also for the backgrounds. Am I missing something? Thanks!